flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java

```diff
@@ -145,8 +145,14 @@ protected Catalog createCatalog(
       baseNamespace = Namespace.of(properties.get(BASE_NAMESPACE).split("\\."));
     }
 
-    boolean cacheEnabled = Boolean.parseBoolean(properties.getOrDefault(CACHE_ENABLED, "true"));
-    return new FlinkCatalog(name, defaultDatabase, baseNamespace, catalogLoader, cacheEnabled);
+    long cacheExpirationIntervalMs =
+        PropertyUtil.propertyAsLong(
+            properties,
+            CatalogProperties.CACHE_EXPIRATION_INTERVAL_MS,
+            CatalogProperties.CACHE_EXPIRATION_INTERVAL_MS_DEFAULT);
+
+    return new FlinkCatalog(
+        name, defaultDatabase, baseNamespace, catalogLoader, cacheExpirationIntervalMs);
```

Comment:
this breaks the compatibility. I feel the benefit probably doesn't justify the breakage. we should keep the boolean config and just support the additional expire interval.